### ISSUE TITLE
New multiline-comment-style fails for jsdoc on export default of objects

### ISSUE THREAD

**Issue by jaapz** [eslint/eslint#9480]
**Tell us about your environment**

* **ESLint Version:** 4.9.0
* **Node Version:** 8.6.0
* **npm Version:** 5.3.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**What did you expect to happen?**
I configured th new rule multiline-comment-style as an error, using "separate-lines", in yaml, like this:

```yaml
    - multiline-comment-style: [2, separate-lines]
```

I'd expect no error on these jsdoc blocks:

```javascript
import {Model} from 'backbone';

/**
 * Some docs
 *
 * @class SomeModel
 */
export default Model.extend({
    defaults: {
        'some': 'fields'
    }
});
```

I tried this, but this also fails in the same way:

```javascript
import {Model} from 'backbone';

/**
 * Some docs
 *
 * @class SomeModel
 */
const SomeModel = Model.extend({
    defaults: {
        'some': 'fields'
    }
});

export default SomeModel;
```

**What actually happened? Please include the actual, raw output from ESLint.**

Got an error on the line containing the comment, saying "Expected multiple line comments instead of a block comment.".

It does not give the same error for functions defined like this:

```javascript
/**
 * My function
 *
 * @function myFunc
 */
const myFunc = function() {

}
```

Or this:

```javascript
/**
 * My function
 *
 * @function myFunc
 */
export default function() {

}
```

The way of defining models as described above is a pretty standard way of defining backbone models, so it would be great if we were able to support this style as well.

**Comment by platinumazure** [comment]
Wondering if we have a general problem with JSDoc comments not being allowed in the separate-lines configuration of multiline-comment-style...?

**Comment by not-an-aardvark** [comment]
The rule is supposed to ignore all JSDoc comments when in separate-lines (or bare-block) mode, but it's possible there is a bug in that logic.

**Comment by Aladdin-ADD** [comment]
however, I was not able to repro it. see here: [eslint demo](https://eslint.org/demo/#eyJ0ZXh0IjoiLyplc2xpbnQgbXVsdGlsaW5lLWNvbW1lbnQtc3R5bGU6IFsyLCBcInNlcGFyYXRlLWxpbmVzXCJdKi9cclxuLyoqXHJcbiAqIFNvbWUgZG9jc1xyXG4gKlxyXG4gKiBAY2xhc3MgU29tZU1vZGVsXHJcbiAqL1xyXG4iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjYsInNvdXJjZVR5cGUiOiJtb2R1bGUiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnsiY29uc3RydWN0b3Itc3VwZXIiOjIsIm5vLWNhc2UtZGVjbGFyYXRpb25zIjoyLCJuby1jbGFzcy1hc3NpZ24iOjIsIm5vLWNvbXBhcmUtbmVnLXplcm8iOjIsIm5vLWNvbmQtYXNzaWduIjoyLCJuby1jb25zb2xlIjoyLCJuby1jb25zdC1hc3NpZ24iOjIsIm5vLWNvbnN0YW50LWNvbmRpdGlvbiI6Miwibm8tY29udHJvbC1yZWdleCI6Miwibm8tZGVidWdnZXIiOjIsIm5vLWRlbGV0ZS12YXIiOjIsIm5vLWR1cGUtYXJncyI6Miwibm8tZHVwZS1jbGFzcy1tZW1iZXJzIjoyLCJuby1kdXBlLWtleXMiOjIsIm5vLWR1cGxpY2F0ZS1jYXNlIjoyLCJuby1lbXB0eS1jaGFyYWN0ZXItY2xhc3MiOjIsIm5vLWVtcHR5LXBhdHRlcm4iOjIsIm5vLWVtcHR5IjoyLCJuby1leC1hc3NpZ24iOjIsIm5vLWV4dHJhLWJvb2xlYW4tY2FzdCI6Miwibm8tZXh0cmEtc2VtaSI6Miwibm8tZmFsbHRocm91Z2giOjIsIm5vLWZ1bmMtYXNzaWduIjoyLCJuby1nbG9iYWwtYXNzaWduIjoyLCJuby1pbm5lci1kZWNsYXJhdGlvbnMiOjIsIm5vLWludmFsaWQtcmVnZXhwIjoyLCJuby1pcnJlZ3VsYXItd2hpdGVzcGFjZSI6Miwibm8tbWl4ZWQtc3BhY2VzLWFuZC10YWJzIjoyLCJuby1uZXctc3ltYm9sIjoyLCJuby1vYmotY2FsbHMiOjIsIm5vLW9jdGFsIjoyLCJuby1yZWRlY2xhcmUiOjIsIm5vLXJlZ2V4LXNwYWNlcyI6Miwibm8tc2VsZi1hc3NpZ24iOjIsIm5vLXNwYXJzZS1hcnJheXMiOjIsIm5vLXRoaXMtYmVmb3JlLXN1cGVyIjoyLCJuby11bmRlZiI6Miwibm8tdW5leHBlY3RlZC1tdWx0aWxpbmUiOjIsIm5vLXVucmVhY2hhYmxlIjoyLCJuby11bnNhZmUtZmluYWxseSI6Miwibm8tdW5zYWZlLW5lZ2F0aW9uIjoyLCJuby11bnVzZWQtbGFiZWxzIjoyLCJuby11bnVzZWQtdmFycyI6Miwibm8tdXNlbGVzcy1lc2NhcGUiOjIsInJlcXVpcmUteWllbGQiOjIsInVzZS1pc25hbiI6MiwidmFsaWQtdHlwZW9mIjoyLCJnZXR0ZXItcmV0dXJuIjoyLCJtdWx0aWxpbmUtY29tbWVudC1zdHlsZSI6Mn0sImVudiI6eyJub2RlIjp0cnVlfX19)

**Comment by jaapz** [comment]
That's weird, I can't reproduce it in your demo thing either, I'll check later today if I can still reproduce it on my local install.

**Comment by jaapz** [comment]
Very sorry for wasting all you guy's times, bug was in our own code as the jsdoc was aligned wrong:

```javascript
/**
* There was no space before this line, while it should have one
*/
export default Model.extend({
 ...
});
```

Eslint sees this (i think correctly) as a block comment.
